### Importar librerias

In [28]:
import numpy as np
import pandas as pd

from scipy.stats import norm

### Cargar datos

In [39]:
data = [[650,730,510,670,480,800,690,530,590,620,710,670,640,780,650,490,800,600,510,700],
        [630,720,462,631,440,783,673,519,543,579,677,649,632,768,615,463,781,563,488,650]]
data = np.array(data).T
df = pd.DataFrame(data, columns = ["girls", "boys"])
df.head()

girls  boys
0    650   630
1    730   720
2    510   462
3    670   631
4    480   440

### Calcular Z-score

In [40]:
alpha = 0.05
mu_diff = 10
sigma_girls = 100 # population, not sample
sigma_boys = 90 # population, not sample
mean_girls = df.loc[:, "girls"].mean()
mean_boys = df.loc[:, "boys"].mean()
n_girls = df.shape[0]
n_boys = df.shape[0]
z_score = (np.abs(mean_girls-mean_boys) - mu_diff)/np.sqrt((sigma_girls**2/n_girls)+(sigma_boys**2/n_boys))
z_score

0.5883679083533637

<h1 align="center">Tabla</h1>

</center><img src="http://www.z-table.com/uploads/2/1/7/9/21795380/9340559_orig.png" width=800></center>

La tabla en este caso particular, es la de la cola izquierda para una distribución con media 0. Por eso los valores negativos. La tabla de la cola derecha es igual solo que positiva.

En la tabla el eje vertical contiene las primeras dos cifras decimales del `z-score`. El eje horizontal tenemos la tercer cifra decimal del `z-score`. Por ejemplo, para ver el `p-value` asociado al `z-score` = 0.588, primero vemos el eje vertical y vamos a `z-score`= 0.5. Luego, en el eje horizontal tenemos los valores 0.08 y 0.09. Hay que hacer una interpolación lineal entre estos dos valores para tener el valor asociado a 0.588. Para `z-score`= 0.58 tenemos que el `p-value` es 0.2810 y para `z-score`= 0.59 tenemos que el `p-value` es 0.2776. Entonces el valor para `z-score`= 0.588 es:

$$ z-score = \frac{(0.59-0.588)}{(0.59-0.58)}*0.2810 + \frac{(0.588-0.58)}{(0.59-0.58)}*0.2776 = 0.27828  $$

In [29]:
def twoSampZ(X1, X2, mudiff, sd1, sd2, n1, n2):
    pooledSE = np.sqrt(sd1**2/n1 + sd2**2/n2)
    z = ((X1 - X2) - mudiff)/pooledSE
    pval = 2*(1 - norm.cdf(abs(z)))
    return np.round(z, 3), np.round(pval, 4)

In [31]:
z, p = twoSampZ(df.loc[:,"girls"], df.loc[:,"boys"], mu_diff, sigma_girls, sigma_boys, n_girls, n_boys)
result = pd.DataFrame()

(0     0.332
 1     0.000
 2     1.263
 3     0.964
 4     0.997
 5     0.233
 6     0.233
 7     0.033
 8     1.230
 9     1.030
 10    0.765
 11    0.366
 12   -0.066
 13    0.066
 14    0.831
 15    0.565
 16    0.299
 17    0.898
 18    0.399
 19    1.330
 dtype: float64,
 array([0.7396, 1.    , 0.2065, 0.335 , 0.3187, 0.816 , 0.816 , 0.9735,
        0.2187, 0.3028, 0.4445, 0.7146, 0.947 , 0.947 , 0.406 , 0.572 ,
        0.7648, 0.3694, 0.69  , 0.1836]))

In [ ]:
0.265695
0.2998